application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -131,6 +143,90 @@ private RestAction<Message> sendExplanationMessage(GuildMessageChannel threadCha
         return action.setEmbeds(embeds);
     }
 
+    /**
+     * Determine between the title of the thread and the first message which to send to the AI. It
+     * uses a simple heuristic of length to determine if enough context exists in a question. If the
+     * title is used, it must also include a question mark since the title is often used more as an
+     * indicator of topic versus a question.
+     * 
+     * @param questionFirstMessage - The first message of the thread which originates from the
+     *        question asker.
+     * @param threadChannel - The thread in which the question was asked.
+     * @return An answer for the user from the AI service or a message indicating either an error or
+     *         why the message wasn't used.
+     */
+    RestAction<Message> constructChatGptAttempt(ThreadChannel threadChannel,
+            String questionFirstMessage) {
+
+        Optional<String> questionOptional =
+                prepareChatGptQuestion(threadChannel, questionFirstMessage);
+        Optional<String> chatGPTAnswer;
+
+        try {
+            String question = questionOptional.orElseThrow();
+            logger.debug("The final question sent to chatGPT: {}", question);
+            chatGPTAnswer = chatGptService.ask(question);
+
+            String aiResponse = chatGPTAnswer.orElseThrow();
+            String lowercaseAiResponse = aiResponse.toLowerCase(Locale.US);
+            if (lowercaseAiResponse.contains("as an ai language model")) {
+                logger.debug("Response from ChatGPT, lacks context or unable to respond?: {}",
+                        aiResponse);
+                return useChatGptFallbackMessage(threadChannel);
+            }
+
+            UnaryOperator<String> response =
+                    """
+                            Here is an AI assisted attempt to answer your question 🤖. Maybe it helps! \
+                            In any case, a human is on the way 👍. To continue talking to the AI, you can use \
+                            %s.
+                            """::formatted;
+
+            return mentionGuildSlashCommand(threadChannel.getGuild(), ChatGptCommand.COMMAND_NAME)
+                .map(response)
+                .flatMap(threadChannel::sendMessage)
+                .flatMap(embed -> threadChannel
+                    .sendMessageEmbeds(HelpSystemHelper.embedWith(aiResponse)));
+        } catch (NoSuchElementException e) {
+            logger.warn("Something went wrong while trying to communicate with the ChatGpt API: {}",
+                    e.getMessage());
+            return useChatGptFallbackMessage(threadChannel);
+        }
+    }
+
+    private Optional<String> prepareChatGptQuestion(ThreadChannel threadChannel,
+            String questionFirstMessage) {
+        String questionTitle = threadChannel.getName();
+        StringBuilder questionBuilder = new StringBuilder(MAX_QUESTION_LENGTH);
```

Comment:
i suppose questions are rarely max length, are they? try to go for something that is more common in practice, maybe `max / 2`?

Reply:
You would be surprised just how fast 200 characters gets filled up. Just taking some sample questions that are on the forum right now they all are around 200 or exceed 200 characters (MAX_LENGTH as defined of now). If the person includes almost any code (and with Java being verbose) the characters are destroyed. It might even be good if we carve out code in the future and just try to get to the human language bits of the question.

The other bit is that I add the tags to the question if a question is too short thus pushing up question lengths.